### ISSUE TITLE
color contrast error: Unable to determine contrast ratio

### ISSUE THREAD

**Issue by connorjclark** [dequelabs/axe-core#2178]
https://www.coursehero.com/file/11/030507/

I bisected the issue somewhere between 3.4.2 and 3.5.0.

![image](https://user-images.githubusercontent.com/4071474/80293516-e2a85600-8714-11ea-8539-56d00074ef16.png)

![image](https://user-images.githubusercontent.com/4071474/80293501-cad0d200-8714-11ea-8198-c39dcd318a3c.png)

**Comment by straker** [comment]
There's a lot of incomplete color-contrast on that page. Are we just talking about the first one specifically with the breadcrumb? If so I would say that's a bug in our algorithm caused by the text truncation.

I look at the others and they are reporting issues due to the large full page image on the page that is over the content. Those are correctly being identified as incomplete.

**Comment by connorjclark** [comment]
Sorry for the poor repro. It happens in Lighthouse: #10644

I've isolated it:

1. Set viewport to 368x500 in DevTools
2. inject axe v3.5.1 in console
3. run

```
window.axe.run(document, {
    elementRef: true,
    runOnly: {
      type: 'tag',
      values: [
        'wcag2a',
        'wcag2aa',
      ],
    },
    resultTypes: ['violations', 'inapplicable'],
    rules: {
      
    },
    // @ts-ignore
  }).then(console.log)
```

You'll see the error:

```
TypeError: Cannot read property 'filter' of undefined
    at D (<anonymous>:12:430410)
    at <anonymous>:12:431399
    at Array.map (<anonymous>)
    at Object.h.getTextElementStack (<anonymous>:12:431376)
    at Object.l.getRectStack (<anonymous>:12:423090)
    at Object.l.filteredRectStack (<anonymous>:12:422635)
    at Object.l.getBackgroundStack (<anonymous>:12:421999)
    at Object.l.getBackgroundColor (<anonymous>:12:421495)
    at Object.evaluate (<anonymous>:12:344532)
    at i.run (<anonymous>:12:152848)
    at <anonymous>:12:154678
    at <anonymous>:12:217962
    at Object.defer (<anonymous>:12:217996)
    at <anonymous>:12:154656
    at Array.forEach (<anonymous>)
    at b.runChecks (<anonymous>:12:154566)
    at <anonymous>:12:155580
    at <anonymous>:12:217962
    at Object.defer (<anonymous>:12:217996)
    at <anonymous>:12:155558
    at Array.forEach (<anonymous>)
    at <anonymous>:12:155536
    at <anonymous>:12:217962
    at Object.defer (<anonymous>:12:217996)
    at <anonymous>:12:155471
    at Array.forEach (<anonymous>)
    at b.run (<anonymous>:12:155449)
    at <anonymous>:12:140996
    at <anonymous>:12:217962
    at Object.defer (<anonymous>:12:217996)
    at <anonymous>:12:149548
    at Array.forEach (<anonymous>)
    at r.run (<anonymous>:12:149526)
    at <anonymous>:12:170948
    at <anonymous>:12:217962
    at Object.defer (<anonymous>:12:217996)
    at Object.$ [as _runRules] (<anonymous>:12:170878)
    at Object.axe.run (<anonymous>:12:174065)
    at <anonymous>:1:12
```

**Comment by paulirish** [comment]
@connorjclark  is this not reproducible in 3.5.0?

**Comment by connorjclark** [comment]
It happens in 3.5.0+.

**Comment by patrickhulce** [comment]
This also happens on https://www.etsy.com/ using just the axe extension without Lighthouse involved.

**Comment by straker** [comment]
This seems to be the same issue as the breadcrumb in https://github.com/dequelabs/axe-core/issues/2178 where text truncation is causing some interesting problems.

**Comment by padmavemulapati** [comment]
Verified with axe-core 3.5.5 ,not seeing any error and its showing results in passes:
![image](https://user-images.githubusercontent.com/47409579/85301344-540e4600-b4c5-11ea-9992-20811f23e40f.png)

**Comment by jeankaplansky** [comment]
no user-facing docs required.
